text/0000-macro-lifetimes.md

```diff
@@ -0,0 +1,49 @@
+- Feature Name: Allow `lifetime` specifiers to be passed to macros
+- Start Date: 2016-04-22
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a `lifetime` specifier for `macro_rules!` patterns, that matches any valid
+lifetime.
+
+# Motivation
+[motivation]: #motivation
+
+Certain classes of macros are completely impossible without the ability to pass
+lifetimes. Specifically, anything that wants to implement a trait from inside of
+a macro is going to need to deal with lifetimes eventually. They're also
+commonly needed for any macros that need to deal with types in a more granular
+way than just `ty`.
+
+Since a lifetime is a single token, there is currently no way to accept one
+without an explicit matcher. Something like `'$lifetime:ident` will fail to
+compile.
```

Comment:
Gotcha. Thanks for the link. I'm also interested because I would like to use this in Diesel as a workaround until this (hopefully) is accepted and in stable